homeassistant/components/matter/entity.py

```diff
@@ -62,9 +62,13 @@ def __init__(
         server_info = matter_client.server_info
         # The server info is set when the client connects to the server.
         assert server_info is not None
+        # create unique_id based on "Operational Instance Name" and endpoint/device type
+        fab_id_hex = f"{server_info.compressed_fabric_id:016X}"
+        node_id_hex = f"{node.node_id:016X}"
+        operational_instance_name = f"{fab_id_hex.upper()}-{node_id_hex.upper()}"
```

Comment:
Upper should be guaranteed by the capital `X` in the formatting string.  So this can be:
```suggestion
        fab_id_hex = f"{server_info.compressed_fabric_id:016X}"
        node_id_hex = f"{node.node_id:016X}"
        operational_instance_name = f"{fab_id_hex}-{node_id_hex}"
```